website/server/models/challenge.js

```diff
@@ -149,7 +154,7 @@ schema.methods.syncToUser = async function syncChallengeToUser (user) {
     }
   });
 
-  toSave.push(user.save());
+  toSave.push(User.collection.update({_id: user._id}, user.toObject()));
```

Comment:
Another thing: `update` doesn't return a promise unless you call `.exec()` at the end http://mongoosejs.com/docs/promises.html